service/src/main/java/uk/gov/hmcts/reform/fpl/service/ActionCmoService.java

```diff
@@ -50,4 +60,17 @@ public void progressCMOToAction(CaseDetails caseDetails, CaseManagementOrder ord
                                                                        List<Element<HearingBooking>> hearingDetails) {
         return draftCMOService.extractIndividualCaseManagementOrderObjects(order, hearingDetails);
     }
+
+    public DocmosisDocument getDocmosisDocument(CaseData data, boolean approved) throws IOException {
+        Map<String, Object> cmoDocumentTemplateData = cmoDocmosisTemplateDataGenerationService.getTemplateData(data,
+            approved);
```

Comment:
Looks like something's wrong here. You're passing `approved` argument to `getTemplateData`, where it's named `draft` and used to put watermark if true. And at the same time, you're putting "draft-" prefix when it's false on line 71.

Reply:
this has been totally refactored and my branch is behind. Will update.